wordpress-mobile/WordPress-iOS
# Remove WordPress widget and iOS 13 Today widget related code

Fixes #21519 

## Description

### WordPress widget-related removals
- Removing WordPressStatsWidgets target, since we no longer support widgets on the WordPress app
- Removing WordPressIntents target, it's used for selecting a site for a widget - https://github.com/wordpress-mobile/WordPress-iOS/pull/15397
- Removing target related code from Podfile
- Removing target related code from Fastfile
- Renaming WordPressStatsWidgets and WordPressIntents folders to JetpackStatsWidgets and JetpackIntents, updating places where the names are referenced (I looked for occurrences with grep)

### iOS 13 widget-related removals
Targets and some of the code was removed with https://github.com/wordpress-mobile/WordPress-iOS/pull/19509, however, some of the code was still in the codebase and it created confusion whether it referenced iOS14 or iOS13 widgets

- Removing constants
- Removing stats and site information saving logic (iOS 14 widgets use different logic)
- Removing Today extension-related code

## To test:

1. Open Jetpack
2. Login
3. Go to home screen, long tap, and select Jetpack
4. Add widget
5. Confirm information appears for a site
6. Long-hold widget
7. Confirm site can be changed
8. Change the site
9. Confirm that the data loads

## Regression Notes
1. Potential unintended areas of impact

Breaking current widget behavior, breaking fastlane flows, breaking localization flows

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing, CI, and code-reviews

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:
